### ISSUE TITLE
Back button returns to registration screen

### ISSUE THREAD

**Issue by Sweater-Baron** [WhisperSystems/Signal-Android#5953]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
After installing signal, I couldn't use SMS verification, so I used the phone call verification. After completing the registration process, I noticed that hitting the phone's back button from the main Signal screen returned me to the SMS verification screen.

### Steps to reproduce
- Install Signal
- When registering Signal, allow the SMS verification to fail, and then use phone call verification
- After registration completes and you are in the main part of the Signal app, hit the phone's back button

**Actual result:** You are returned to the registration screen
**Expected result:** Navigating to a part of the registration screen should not be possible after Signal is registered.

### Screenshots

The screen that the back button returns me to:
![screenshot_20161212-211252](https://cloud.githubusercontent.com/assets/9122156/21129117/97c81d02-c0b4-11e6-8db9-5be3716c2d4b.png)

### Device info
**Device:** Google Pixel XL
**Android version:** 7.1
**Signal version:** 3.24.1

### Link to debug log
https://gist.github.com/anonymous/dfc997058c134a75be9e24de5805f481

**Comment by 2-4601** [comment]
@Sweater-Baron  Can you verify if this is a duplicate of #4928?

**Comment by Sweater-Baron** [comment]
That's similar, but I'm not actually un-registered. Pressing the home button and re-opening the app brought me back to the regular Signal screen.

**Comment by Sweater-Baron** [comment]
It looks like I did become un-registered at some point, sort of. Now I'm in a state where I can receive encrypted messages, but I can't send them because the app thinks I'm not registered. So this probably is a duplicate of #4928 after all.

Here is another debug log: https://gist.github.com/anonymous/ee081481f816cfcc39a60a90972a107e

Another thing worth noting is that this happened after I got a new phone, and I never uninstalled Signal from my old phone. I don't know if that would affect anything or not.

![unnamed](https://cloud.githubusercontent.com/assets/9122156/21204959/a14755d6-c20e-11e6-957a-85a84d9584dd.png)
![unnamed 1](https://cloud.githubusercontent.com/assets/9122156/21204958/a144abe2-c20e-11e6-9746-b2873fd3eb83.png)
